common/persistence/task_manager.go

```diff
@@ -139,6 +139,17 @@ func (m *taskManagerImpl) GetTaskQueue(
 	if err != nil {
 		return nil, err
 	}
+
+	numTasks, err := m.taskStore.CountTasksFromTaskQueue(ctx, &CountTasksFromTaskQueueRequest{
```

Comment:
This is a smart approach but I'm worried about the additional load that it imposes on DB when loading task queues. Consider matching restart when suddenly many task queues have to reload in another node. 

IMO we should call CountTasksFromTaskQueue only when the count is needed because user asked for it (i.e. backlog info is requested).

That probably means that the logic for choosing between the counter or CountTasksFromTaskQueue should be pulled up to backlog manager itself.

Reply:
Agreed, thanks for the pointer. Changes have been made to now have the logic encapsulated inside of `physicalTaskQueueManager` since I believe this is where we check if a user has requested backlog information or not.